### PR TITLE
Change contributors list

### PR DESCRIPTION
**Pull request by richard015ar**
Issue: https://github.com/pressbooks/pressbooks/issues/2231

This PR changes the Contributors list in Admin Book > Book Info > Contributors, and displays 4 columns:
- Picture
- Name (sortable)
- Institution (sortable)
- Biographical Info (sortable): Max 180 characteres

It also fixes: https://github.com/pressbooks/pressbooks/issues/2355

### Test
- Go to Admin Book > Book Info > Contributors and make sure the tables mentioned above are displayed in the order proposed.
- Make sure the sort actions works for all columns (except the Picture one, it should be disabled).

#### For https://github.com/pressbooks/pressbooks/issues/2355
- Edit a contributor with a profile picture.
- Change one or more fields, except for the picture one and save it.
- Make sure the contributor still has the profile picture in place.

### CODE REVIEW

File: inc/admin/metaboxes/namespace.php [hunk 1]
```diff
@@ -1273,12 +1273,20 @@ function contributor_edit_form( $term ) {
 						<td>
 							<?php if ( $value ) : ?>
 								<img src="<?php echo $value; ?>" id="<?php echo $meta_tags['tag']; ?>-thumbnail" width="120" /> <br />
+							<?php else : ?>
+								<img style="display: none" id="<?php echo $meta_tags['tag']; ?>-thumbnail" width="120" />
 							<?php endif; ?>
+							<br />
```
Comment (richard015ar) [inline inc/admin/metaboxes/namespace.php:1279]: This will fix: https://github.com/pressbooks/pressbooks/issues/2355

Comment (fdalcin) [reply inc/admin/metaboxes/namespace.php:1279]: Tested and it's working as expected now

File: inc/metadata/namespace.php [hunk 2]
```diff
@@ -1016,6 +1017,36 @@ function register_contributor_meta() {
...
+
+	if (
+		$pagenow === 'edit-tags.php' &&
+		isset( $args['taxonomy'] ) &&
+		isset( $args['orderby'] ) &&
+		in_array( Contributors::TAXONOMY, $args['taxonomy'], true ) &&
+		in_array( $args['orderby'], $order_field_allowed, true )
+	) {
+		global $wpdb;
+		$pieces['join'] .= ' INNER JOIN ' . $wpdb->termmeta . ' AS tm ON t.term_id = tm.term_id ';
```
Comment (fdalcin) [inline inc/metadata/namespace.php:1043]: Maybe we should use a LEFT JOIN here to avoid removing contributors that don't have a field.

**Review by fdalcin — CHANGES_REQUESTED** [review]
LGTM @richard015ar 

Two things that I noticed.

1. We could reduce the picture size to 64px maybe and have borders rounded to have a better UI.
image

1. Sorting the list works properly, except when the field is empty. When sorting all contributors that don't have that field will be removed from the list, see below:

https://user-images.githubusercontent.com/1761690/132014095-a070ff49-6780-401c-90ad-17c5b77445ce.mov

File: inc/metadata/namespace.php (hunk 2 above)
Comment (richard015ar) [reply inc/metadata/namespace.php:1043]: oh yes, you're right! I will try with LEFT JOIN and that should solve the issue. Thanks for catching it!

**Comment by richard015ar** [comment]
@fdalcin regarding to point 1, would be nice to ask to @SteelWagstaff to confirm the UI. Honestly, I don't know what's better to UX and UI level, your suggestion looks nicer to me, but I would like to confirm this suggestion with Steel.

Thank you for catching the 2 point, I will try with the LEFT JOIN to solve it.

**Comment by richard015ar** [comment]
@fdalcin the sorting feature for custom columns in the contributors list has been removed, since a custom query in a different kind of hook would be required. LEFT JOIN is useful but not enough, since the WHERE condition restricts the ordering.
Only way to order by a specific meta_value is by using WP_Query object, which requires other handling.
@SteelWagstaff mentioned this is not a required feature, although I tried to add it, after 2 hours I decided to keep it out of the scope given the priorities of this sprint.  

I added the style you suggested, now look nicer.

**Review by fdalcin — APPROVED** [review]

**Pull request merged**